### PR TITLE
[rt-smart] 修复 aarch64 timer 地址转换问题

### PR DESCRIPTION
**Pull request by zhkag**
## 拉取/合并请求描述：(PR description)

[

#### 在什么测试环境下测试通过 (what is the test environment)

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: components/libc/compilers/common/ctime.c [hunk 1]
```diff
@@ -969,7 +1031,18 @@ int timer_create(clockid_t clockid, struct sigevent *evp, timer_t *timerid)
         rt_timer_init(&timer->timer, timername, rtthread_timer_wrapper, timer, 0, RT_TIMER_FLAG_ONE_SHOT | RT_TIMER_FLAG_SOFT_TIMER);
     }
 
-    *timerid = (timer_t)((uintptr_t)timer >> 1);
+    timer_id_lock();
+    _timerid = timer_id_alloc();
+    if (_timerid < 0)
+    {
+        LOG_E("_timerid overflow!");
+        return -1; /* todo:memory leak */
```
Comment (geniusgogo) [inline components/libc/compilers/common/ctime.c:1039]: 这里出错返回，锁没释放？

File: components/libc/compilers/common/ctime.c [hunk 2]
```diff
@@ -1020,7 +1105,8 @@ int timer_getoverrun(timer_t timerid)
  */
 int timer_gettime(timer_t timerid, struct itimerspec *its)
 {
-    struct timer_obj *timer = (struct timer_obj *)((uintptr_t)timerid << 1);
+    // struct timer_obj *timer = (struct timer_obj *)((uintptr_t)timerid << 1);
```
Comment (geniusgogo) [inline components/libc/compilers/common/ctime.c:1108]: 无用代码干掉

File: components/libc/compilers/common/ctime.c [hunk 3]
```diff
@@ -982,7 +1055,19 @@ RTM_EXPORT(timer_create);
  */
 int timer_delete(timer_t timerid)
 {
-    struct timer_obj *timer = (struct timer_obj *)((uintptr_t)timerid << 1);
+    // struct timer_obj *timer = (struct timer_obj *)((uintptr_t)timerid << 1);
+    struct timer_obj *timer;
+    timer_id_lock();
+    if (_g_timerid[(int)timerid] == NULL)
+    {
+        LOG_E("can not find timer!");
```
Comment (geniusgogo) [inline components/libc/compilers/common/ctime.c:1063]: 锁里面经量不要加打印

**Comment by BernardXiong** [comment]
一般环境也可用？例如stm32，或许后面需要多检查下qemu stm32的环境。

File: components/libc/compilers/common/ctime.c [hunk 4]
```diff
@@ -918,6 +918,67 @@ static void rtthread_timer_wrapper(void *timerobj)
 #endif
 }
 
+#define TIMER_ID_MAX 50
```
Comment (geniusgogo) [inline components/libc/compilers/common/ctime.c:921]: 这个可否添加kconfig配置？

**Review by geniusgogo — APPROVED** [review]

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**